src/library/features/crates/cratetracks.h

```diff
@@ -0,0 +1,83 @@
+#ifndef MIXXX_CRATETRACKS_H
+#define MIXXX_CRATETRACKS_H
+
+#include <QObject>
+#include <QList>
+#include <QSet>
+
+#include "library/dao/dao.h"
+#include "library/crate/cratesummary.h"
+#include "library/features/crates/cratestoragehelpers.h"
+
+#include "util/db/fwdsqlqueryselectresult.h"
+#include "util/db/sqlsubselectmode.h"
+
+class CrateTracks : public virtual DAO {
+  public:
+    CrateTracks() {}
+    ~CrateTracks() override {}
+
+    void initialize(const QSqlDatabase& database) override;
+
+    /////////////////////////////////////////////////////////////////////////
+    // Crate write operations (transactional, non-const)
+    //
+    // Naming conventions:
+    //  on<present participle>...()
+    //    - Invoked within active transaction
+    //    - May fail
+    //    - Performs only database modifications that are either committed
+    //      or implicitly reverted on rollback
+    //  after<present participle>...()
+    //    - Invoked after preceding transaction has been committed (see above)
+    //    - Must not fail
+    //    - Typical use case: Update internal caches and compute change set
+    //      for notifications
+    /////////////////////////////////////////////////////////////////////////
+
+    bool onAddingCrateTracks(
```

Comment:
We don't need to stick to the naming conventions that I once proposed, please delete the comments. I don't consider them useful anymore. DAOs should be statelesss and don't need to deal with transactions from an outer context. A name like addTracksToCrate() for this function should be fine.